junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/TimeoutExtensionTests.java

```diff
@@ -105,25 +105,29 @@ void appliesTimeoutOnAnnotatedTestFactoryMethods() {
 				.hasMessage("testFactoryMethod() timed out after 10 milliseconds");
 	}
 
-	@Test
+	@TestFactory
 	@DisplayName("is applied on testable methods in annotated classes")
-	void appliesTimeoutOnTestableMethodsInAnnotatedClasses() {
-		EngineExecutionResults results = executeTests(request() //
-				.selectors(selectClass(TimeoutAnnotatedClassTestCase.class)) //
-				.configurationParameter(DEFAULT_TEST_METHOD_TIMEOUT_PROPERTY_NAME, "42ns") //
-				.configurationParameter(DEFAULT_TEST_TEMPLATE_METHOD_TIMEOUT_PROPERTY_NAME, "42ns") //
-				.configurationParameter(DEFAULT_TEST_FACTORY_METHOD_TIMEOUT_PROPERTY_NAME, "42ns") //
-				.build());
-
-		Stream.of("testMethod()", "repetition 1", "repetition 2", "testFactoryMethod()").forEach(displayName -> {
-			Execution execution = findExecution(results.all(), displayName);
-			assertThat(execution.getDuration()) //
-					.isGreaterThanOrEqualTo(Duration.ofMillis(10)) //
-					.isLessThan(Duration.ofSeconds(1));
-			assertThat(execution.getTerminationInfo().getExecutionResult().getThrowable().orElseThrow()) //
-					.isInstanceOf(TimeoutException.class) //
-					.hasMessageEndingWith("timed out after 10000000 nanoseconds");
-		});
+	Stream<DynamicTest> appliesTimeoutOnTestableMethodsInAnnotatedClasses() {
+		return Stream.of(TimeoutAnnotatedClassTestCase.class, InheritedTimeoutAnnotatedClassTestCase.class).map(
```

Comment:
@marcphilipp I wonder if it would be feasible to use `@ParameterizedTest` here? Or would it cause a package or dependency cycle, or is it tricky to parameterize on instances of type `Class`? :)

Reply:
> I wonder if it would be feasible to use `@ParameterizedTest` here? Or would it cause a package or dependency cycle, 

That results in a project cycle in Eclipse, since we have not configured Eclipse's (relatively) new support for separating test sources/dependencies.

> or is it tricky to parameterize on instances of type `Class`? :)

That's actually easy: `@ValueSource(classes = { ... })`. In fact, I just used that feature today: https://github.com/spring-projects/spring-framework/commit/50cc23ca55293036e950260b8a90f4b4a3472fff#diff-442e1385f74c82823d0c9520ec395febR70

Reply:
Ah cool, thanks for enlightening me regarding both Eclipse and classes in annotations, @sbrannen! :+1: